nvim/after/ftplugin/tex.lua

```diff
@@ -56,13 +52,7 @@ require("nvim-surround").buffer_setup({
       delete = "^(\\%a-tt{)().-(})()$",
     },
     ["$"] = {
-        add = { "$ ", " $" },
-        find = function()
-            return M.get_selection({ motion = "a$" })
-        end,
-        delete = "^($)().-($)()$",
```

Comment:
If you would like to add the extra space here, then leave the `add` key alone; otherwise you can omit the `$` surround altogether. The delete key here is omitted because VimTeX already provides the `ds$` mapping for you.